pages/food-groups/index.tsx

```diff
@@ -0,0 +1,280 @@
+      )
+    if (!draggableZoneWheelRef.current)
+      throw new Error('Wheel Ref did not return current')
+    setDraggableZone(draggableZoneRef.current.getBoundingClientRect())
+
+    window.addEventListener('resize', () => {
+      if (!draggableZoneRef.current)
+        throw new Error(
+          'reference to parent div containing foodgroup and draggables didnt return .current'
+        )
+      setDraggableZone(draggableZoneRef.current.getBoundingClientRect())
+    })
+
+    window.addEventListener('resize', () => {
+      if (!draggableZoneWheelRef.current)
+        throw new Error('draggableZoneWheelRef not initialised?')
+    })
+  }, [])
+
+  useEffect(() => {
+    if (user && roundCounterSignedOut > 0) {
+      updateAchievementsDocument?.({
+        [ACHIEVEMENT.DRAG_DROP_WIN_COUNT]:
+          achievements[ACHIEVEMENT.DRAG_DROP_WIN_COUNT] + roundCounterSignedOut,
+        // For now, achivements is just the amount of game wins
+        [ACHIEVEMENT.ACHIEVEMENT_COUNT]:
+          achievements[ACHIEVEMENT.ACHIEVEMENT_COUNT] + roundCounterSignedOut
+      })
+      setRoundCounterSignedOut(0) // Reset count so it doesn't get 'double added' if a person relogs
+    }
+  }, [roundCounterSignedOut, achievements]) // Do not include user - user triggers update BEFORE achievements is updated with online data
+
+  const endDragF = (index: number) => {
+    if (hoverType === selectedDraggableType && hoverType != GROUPS.DEFAULT) {
+      const draggingStates_ = [...draggingStates]
+      draggingStates_[index] = DRAGGING_STATE.WHEEL
+      setDraggingStates(draggingStates_)
+      correctDraggables[index] = true // CORRECT ANSWER
+    } else {
+      //if (hoverType !== GROUPS.NONE) { // Dropping on 'nothing' makes it return now.
+      const draggingStates_ = [...draggingStates]
+      draggingStates_[index] = DRAGGING_STATE.START
+      setDraggingStates(draggingStates_)
+      correctDraggables[index] = false // WRONG ANSWER
+      // RESET POSITION
+      if (switchSetFlag)
+        draggablePositions_1[index][1](
+          CHARACTER_POSITIONS[currentCharSet[index].start_index]
+        )
+      else
+        draggablePositions_2[index][1](
+          CHARACTER_POSITIONS[nextCharSet[index].start_index]
+        )
+    }
+
+    // CHECK FOR END OF ROUND
+    if (correctDraggables.every((v: boolean) => v)) {
+      // END OF ROUND
+      setRoundCounter(roundCounter + 1)
+      if (!user) {
+        setRoundCounterSignedOut(roundCounterSignedOut + 1)
+      }
+      updateAchievementsDocument?.({
+        // If achievements based on the number of wins should be implemented in the future.
+        [ACHIEVEMENT.DRAG_DROP_WIN_COUNT]:
+          achievements[ACHIEVEMENT.DRAG_DROP_WIN_COUNT] + 1,
+        // For now, achivements is just the amount of game wins
+        [ACHIEVEMENT.ACHIEVEMENT_COUNT]:
+          achievements[ACHIEVEMENT.ACHIEVEMENT_COUNT] + 1
+      })
```

Comment:
There are some major changes made to `FirebaseContext`. Please update the codes that are using the values and functions from this context.

Reply:
Done